AutoRest/Generators/Java/Azure.Java.Tests/src/main/java/fixtures/paging/PagingOperationsImpl.java

```diff
@@ -433,14 +439,15 @@ public void onResponse(Call<ResponseBody> call, Response<ResponseBody> response)
         Integer timeout = pagingGetMultiplePagesWithOffsetOptions.getTimeout();
         Call<ResponseBody> call = service.getMultiplePagesWithOffset(offset, clientRequestId, this.client.getAcceptLanguage(), maxresults, timeout);
         ServiceResponse<PageImpl<Product>> response = getMultiplePagesWithOffsetDelegate(call.execute());
-        List<Product> result = response.getBody().getItems();
-        while (response.getBody().getNextPageLink() != null) {
-            PagingGetMultiplePagesWithOffsetNextOptions pagingGetMultiplePagesWithOffsetNextOptions = new PagingGetMultiplePagesWithOffsetNextOptions();
-            pagingGetMultiplePagesWithOffsetNextOptions.setMaxresults(pagingGetMultiplePagesWithOffsetOptions.getMaxresults());
-            pagingGetMultiplePagesWithOffsetNextOptions.setTimeout(pagingGetMultiplePagesWithOffsetOptions.getTimeout());
-            response = getMultiplePagesWithOffsetNext(response.getBody().getNextPageLink(), clientRequestId, pagingGetMultiplePagesWithOffsetNextOptions);
-            result.addAll(response.getBody().getItems());
-        }
+        PagedList<Product> result = new PagedList<Product>(response.getBody()) {
+            @Override
+            public Page<Product> loadPage(String nextPageLink) throws CloudException, IOException {
+                PagingGetMultiplePagesWithOffsetNextOptions pagingGetMultiplePagesWithOffsetNextOptions = new PagingGetMultiplePagesWithOffsetNextOptions();
```

Comment:
These 4 lines are called redundantly every time but for now there is no good workaround.
Moving this block out would require `pagingGetMultiplePagesWithOffsetNextOptions` to be declared final but when it's optional we would have something like
```java
final PagingGetMultiplePagesWithOffsetNextOptions pagingGetMultiplePagesWithOffsetNextOptions = null;
if (maxresults != null) {
    pagingGetMultiplePagesWithOffsetNextOptions = new PagingGetMultiplePagesWithOffsetNextOptions();
}
```
which won't compile.